test/cpp/jit/test_schema_info.cpp

```diff
@@ -21,9 +21,38 @@ TEST(FunctionSchemaIsMutableTest, InvalidArgument) {
       torch::jit::getOperatorForLiteral(
           "aten::sub_.Tensor(Tensor(a!) self, Tensor other, *, Scalar alpha=1) -> (Tensor(a!))")
           ->schema();
-  ASSERT_THROW(schema.is_mutable(-1), c10::Error);
   ASSERT_THROW(schema.is_mutable(4), c10::Error);
   ASSERT_THROW(schema.is_mutable("named_argument"), c10::Error);
 }
+
+TEST(FunctionSchemaAreAliasingTest, Basic) {
+  c10::FunctionSchema schema =
+      torch::jit::getOperatorForLiteral(
+          "aten::sub_.Tensor(Tensor(a!) self, Tensor other, *, Scalar alpha=1) -> (Tensor(a!))")
+          ->schema();
+  ASSERT_TRUE(schema.areAliasing({c10::input, 0}, {c10::output, 0}));
+  ASSERT_FALSE(schema.areAliasing({c10::input, 1}, {c10::output, 0}));
+  ASSERT_FALSE(schema.areAliasing({c10::input, 1}, {c10::input, 0}));
+}
+
+TEST(FunctionSchemaAreAliasingTest, InvalidArgument) {
+  c10::FunctionSchema schema =
+      torch::jit::getOperatorForLiteral(
+          "aten::sub_.Tensor(Tensor(a!) self, Tensor other, *, Scalar alpha=1) -> (Tensor(a!))")
+          ->schema();
+  ASSERT_THROW(
+      schema.areAliasing({c10::input, 15}, {c10::output, 0}), c10::Error);
+  ASSERT_THROW(
+      schema.areAliasing({c10::input, 0}, {c10::output, 15}), c10::Error);
+}
+
+TEST(FunctionSchemaAreAliasingTest, Wildcard) {
+  c10::FunctionSchema schema =
+      torch::jit::getOperatorForLiteral(
```

Comment:
fyi, I think you can use this for parsing arbitrary schemas:
https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/frontend/function_schema_parser.h#L13

This might be useful for:
1) parsing fake schemas
2) if you parse fake schemas instead of loading actual operators and getting their schemas, then this test won't break if those ops change.